observer/lib/observer.js

```diff
@@ -0,0 +1,39 @@
+import { updateDailyTransferStats } from './platform-stats.js'
+
+/**
+ * Observe the transfer events on the Filecoin blockchain
+ * @param {import('pg').Pool} pgPool
+ * @param {import('ethers').Contract} ieContract
+ * @param {import('ethers').Provider} provider
+ */
+export const observeTransferEvents = async (pgPool, ieContract, provider) => {
+  const { rows } = await pgPool.query(
+    'SELECT MAX(last_checked_block) FROM daily_reward_transfers'
+  )
+  const lastCheckedBlock = rows[0].last_checked_block
+
+  console.log('Querying impact evaluator Transfer events after block', lastCheckedBlock)
+  let events
+  try {
+    events = await ieContract.queryFilter(ieContract.filters.Transfer(), lastCheckedBlock)
+  } catch (error) {
+    console.error('Error querying impact evaluator Transfer events', error)
+    if (error.message.includes('bad tipset height')) {
```

Comment:
Relying on specific error message text is fragile. A small change in the error message will break our code.

Is there any machine-readable code we could check to detect this error case?